Library/Homebrew/test/rubocops/deprecate_disable/reason_spec.rb

```diff
@@ -324,4 +324,65 @@ class Foo < Formula
       RUBY
     end
   end
+
+  context "when auditing `deprecate!` and `disable!`" do
+    it "reports no offense if deprecate `reason` is absent" do
+      expect_no_offenses(<<~RUBY)
+        class Foo < Formula
+          url 'https://brew.sh/foo-1.0.tgz'
+          disable! date: "2021-08-28", because: :does_not_build
+          deprecate! date: "2020-08-28"
```

Comment:
Not really relevant for this PR, but are these automatically sorted in this order by another cop? To me it would make more sense to have `deprecate!` sorted first.

Reply:
It is from AST order:
https://github.com/Homebrew/brew/blob/1d19230268f8de233b503177f0d4a5a4bb2072c7/Library/Homebrew/ast_constants.rb#L27-L28

Could reverse them if we wanted, though would need to make sure about handling. I did try to add code so order doesn't matter.

Reply:
Will look into order in #18974 prior to continuing this PR.